Snowflake.Client/SnowflakeClient.cs

```diff
@@ -104,10 +141,24 @@ public async Task<bool> InitNewSessionAsync(CancellationToken ct = default)
             return true;
         }
 
+        /// <summary>
+        /// Authenticates user and returns new Snowflake session.
+        /// </summary>
+        /// <returns>New Snowflake session</returns>
         private async Task<SnowflakeSession> AuthenticateAsync(AuthInfo authInfo, SessionInfo sessionInfo, CancellationToken ct)
         {
             var loginRequest = _requestBuilder.BuildLoginRequest(authInfo, sessionInfo);
 
+            if(authInfo is AzureAdAuthInfo azureAdAuthInfo)
```

Comment:
I feel like if alternative providers are within the Authentication loop this will get bloated. Have you considered just taking advantage of `ISnowflakeClient` and then providing an `AzureSnowflakeClient`?